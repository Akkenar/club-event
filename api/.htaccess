<FilesMatch "getRegistrations.php">
  AuthType Basic
  AuthName "Password Protected Area"
  AuthUserFile /home/www/f299233bc103871b19372e6d2c99db54/web/ad-sss/api/.htpasswd
  Require valid-user
</FilesMatch>
