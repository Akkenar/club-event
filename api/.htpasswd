ad-sss:$apr1$G01HZF83$kSFztMQcGPkavFtr07Mng/