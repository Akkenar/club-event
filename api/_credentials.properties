recaptcha.key=TBD
db.host=TBD
db.name=TBD
db.user=TBD
db.password=TBD
email.username=TBD
email.password=TBD
email.host=TBD