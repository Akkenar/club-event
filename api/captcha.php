<?php
require_once './config.php';
require_once './lib/recaptcha-1.2.1/src/autoload.php';

function validateCaptcha($recaptchaResponse)
{
  $config = readConfig();
  if ($config['recaptcha.ignore'] && $recaptchaResponse === 'no-captcha') {
    error_log("Ignoring captcha", 0);
    // By configuration.
    return;
  }

  // Recaptcha utils
  $recaptcha = new \ReCaptcha\ReCaptcha($config['recaptcha.key']);

  // Validate the recaptcha
  $resp = $recaptcha->verify($recaptchaResponse);
  if (!$resp->isSuccess()) {
    $errors = $resp->getErrorCodes();
    http_response_code(500);
    $error_json =
      '{"result": "error", "reason": "' . json_encode($errors) . '"}';
    syslog(LOG_ERR, $error_json);
    echo $error_json;
    die();
  }
}
