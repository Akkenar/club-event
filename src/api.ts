import environment from './environment/environment';

export function getApi(): any {
  // Depending on whether the file is replaced by Webpack, the api URLs
  // will change based on the type of build. See `environment.ts` and
  // `environment.prod.ts` for more details.
  return environment.api;
}
