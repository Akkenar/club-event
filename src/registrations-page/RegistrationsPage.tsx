import * as React from 'react';
import { useContext, useEffect } from 'react';
import { Header } from 'semantic-ui-react';
import getKey from '../core/intl/getKey';
import LanguageContext from '../core/intl/LanguageContext';
import { goToTop, setPageTitle } from '../core/page.lib';
import RegistrationsContainer from './RegistrationsContainer';

const RegistrationsPage = () => {
  const { messages } = useContext(LanguageContext);

  useEffect(() => {
    setPageTitle(getKey('registrations.page.title', messages));
    goToTop();
  });

  return (
    <div className="RegistrationPage">
      <Header as="h1">{getKey('registrations.page.title', messages)}</Header>
      <RegistrationsContainer />
    </div>
  );
};

export default RegistrationsPage;
