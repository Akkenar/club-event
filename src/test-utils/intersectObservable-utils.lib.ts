import { mockGlobalProperty } from './test-utils.lib';

const allObservers: IntersectionObserverMock[] = [];

class IntersectionObserverMock {
  private readonly cb: any;
  public constructor(cb: any) {
    this.cb = cb;

    // To allow global triggering.
    allObservers.push(this);
  }

  public trigger() {
    this.cb([{ isIntersecting: true }]);
  }

  public observe() {
    // Nothing
  }

  public unobserve() {
    // Nothing.
  }

  public disconnect() {
    // Nothing.
  }
}

class IntersectionObserverEntryMock {
  public intersectionRatio() {
    // Nothing.
  }

  public isIntersecting() {
    // Nothing.
  }
}

export function mockInterestObservable(): () => void {
  // Reset all the mocks
  allObservers.slice(0);

  mockGlobalProperty(window)('IntersectionObserver')(IntersectionObserverMock);
  mockGlobalProperty(window)('IntersectionObserverEntry')(
    IntersectionObserverEntryMock,
  );

  // Trigger all.
  return () => {
    allObservers.forEach((observer: IntersectionObserverMock) => {
      observer.trigger();
    });
  };
}
