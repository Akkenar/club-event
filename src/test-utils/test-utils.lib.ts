import { fireEvent } from 'react-testing-library';

export function mockGrecaptcha() {
  (window as any).grecaptcha = {
    getResponse: () => 'response',
    render: jest.fn(),
    reset: jest.fn(),
  };
}

export function mockResponse(response: any) {
  const mock = typeof response === 'string' ? { result: response } : response;

  // Mock the fetch data as JSDom doesn't implement fetch.
  window.fetch = jest.fn(() =>
    Promise.resolve({
      text: () =>
        new Promise(resolve => {
          const mockResult = JSON.stringify(mock);
          setTimeout(() => {
            resolve(mockResult);
          }, 50);
        }),
    })
  );
}

export function setValidInputData(
  validData: any,
  { getByTestId }: { getByTestId: any }
) {
  Object.keys(validData).forEach((key: string) => {
    fireEvent.change(getByTestId(key), {
      // @ts-ignore
      target: { value: validData[key] },
    });
  });
}
