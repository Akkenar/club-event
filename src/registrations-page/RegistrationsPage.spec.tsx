import 'jest-dom/extend-expect';
import * as React from 'react';
import { HashRouter } from 'react-router-dom';
import { render, wait, waitForElement } from 'react-testing-library';
import mockRegistrations from '../../mocks/registrations.json';
import { Results } from '../core/api.type';
import { mockGrecaptcha, mockResponse } from '../test-utils/test-utils.lib';
import RegistrationsPage from './RegistrationsPage';

mockGrecaptcha();

describe('RegistrationsPage', () => {
  it('should match the snapshot', async () => {
    mockResponse(mockRegistrations);
    const wrapper = render(<RegistrationsPage />);
    await waitForElement(() => wrapper.queryByTestId('registrations-total'));
    expect(wrapper.baseElement).toMatchSnapshot();
  });

  it('should handle no registrations', async () => {
    mockResponse([]);
    const wrapper = render(<RegistrationsPage />);
    await waitForElement(() => wrapper.queryByTestId('registrations-total'));
    expect(wrapper.baseElement).toMatchSnapshot();
  });

  it('should set the page title', () => {
    // Will be set by side effect through a hook
    render(<RegistrationsPage />);
    expect(document.title).toEqual('registrations.page.title');
  });

  it('should show the details', async () => {
    mockResponse(mockRegistrations);

    const wrapper = render(<RegistrationsPage />);
    await waitForElement(() => wrapper.queryByTestId('details-1'));

    wrapper.queryByTestId('details-1').click();
    expect(wrapper.queryByTestId('details-for-1')).toBeTruthy();
    expect(wrapper.baseElement).toMatchSnapshot();
  });

  it('should show and hide the details', async () => {
    mockResponse(mockRegistrations);

    const wrapper = render(<RegistrationsPage />);
    await waitForElement(() => wrapper.queryByTestId('details-1'));

    wrapper.queryByTestId('details-1').click();
    expect(wrapper.queryByTestId('details-for-1')).toBeTruthy();

    wrapper.queryByTestId('details-1').click();
    expect(wrapper.queryByTestId('details-for-1')).toBeFalsy();
  });

  it('should show the products', async () => {
    mockResponse(mockRegistrations);

    const wrapper = render(<RegistrationsPage />);
    await waitForElement(() => wrapper.queryByTestId('toggle-products'));

    wrapper.queryByTestId('toggle-products').click();
    expect(wrapper.baseElement).toMatchSnapshot();
  });

  it('should redirect to the login page if not authorized', async () => {
    mockResponse({
      result: Results.UNAUTHORIZED,
    });

    const RegistrationsPageWithRouter = () => (
      <HashRouter>
        <RegistrationsPage />
      </HashRouter>
    );

    const wrapper = render(<RegistrationsPageWithRouter />);

    // First wait for the loader to be displayed
    await waitForElement(() => wrapper.queryByTestId('loader'));

    // Wait for the loader to be dismissed
    await wait(() =>
      expect(wrapper.queryByTestId('loader')).not.toBeInTheDocument()
    );

    expect(window.location.href).toEqual('http://localhost/#/de/login');
  });
});
