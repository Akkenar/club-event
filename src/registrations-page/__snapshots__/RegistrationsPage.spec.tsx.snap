// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationsPage should match the snapshot 1`] = `
<body
  class=""
>
  <div>
    <div
      class="RegistrationPage"
    >
      <h1
        class="ui header"
      >
        registrations.page.title
      </h1>
      <div
        class="Registrations__Total"
        data-testid="registrations-total"
      >
        3
      </div>
    </div>
  </div>
</body>
`;
